docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+
+The interactive `rebase` command finally completes.
+
+![submit_pr_squash_commits_9.png](./imgs/submit_pr_squash_commits_9.png)
+
+
+Now it is time to push the changes from the local feature branch to the remote feature branch.
+
+Type :
+
+```
+git push -f
+```
+
+You can confirm that there is only 1 commit by typing:
+
+```
+git log
+```
+
+![submit_pr_squash_commits_11.png](./imgs/submit_pr_squash_commits_11.png)
+
+or via your browser in your remote feature branch.
+
+![submit_pr_squash_commits_10.png](./imgs/submit_pr_squash_commits_10.png)
+
+
+### 17. Creating the pull request (PR)
+
+Now that you have completed the `rebase` and `commit squashing` steps, it is finally time to create the pull request.
+
+In your browser, navigate to your remote `feature` branch, and press the `Compare & pull request` button.
+
+![submit_pr_create_pr_1.png](./imgs/submit_pr_create_pr_1.png)
+
+The short and long description fields of the pull request are auto-filled using the short and long description of your commit.
+
+![submit_pr_create_pr_2.png](./imgs/submit_pr_create_pr_2.png)
+
+Fill in as much extra information as possible to properly describe the purpose of the pull request ( for example, fixing a specific bug or implementing an extension point, etc).
+
+(The title of the pull request should start with “WIP” for “work in progress” when the pull request is not complete yet.)
+
+ Press the `Create pull request` button.
+
+
+
+![submit_pr_create_pr_3.png](./imgs/submit_pr_create_pr_3.png)
+
+The pull request is created, and some continuous integration jobs commence.
+
+![submit_pr_create_pr_4.png](./imgs/submit_pr_create_pr_4.png)
+
+Make certain all these jobs complete successfully.
+
+### 18. Agreeing to the contributor license agreement (CLA)
+
+You must agree to the contributor license agreement (CLA) before the pull request can be `merged`. Please read
+[Agreeing to the CLA](https://loopback.io/doc/en/contrib/code-contrib.html#agreeing-to-the-cla).
+
+### 19. Go through the PR review process
+
+Request a code review or document review.
+
+Add a specific committer to speed up this process. (for example, @bajtos, @raymondfeng).
+
+The review process is iterative, so it is normal that multiple extra commits may be required until the pull request is finally accepted.
+
+
+### 20. Final rebase and squashing of commits
+
+Once the pull request is finally approved, repeat the `Rebase` and the `Squashing Commits` sections so that your pull request has a single, clean final commit and corresponding message.
+
+Ensure all the continuous integration (CI) jobs pass successfully.
+
+![submit_pr_final_commit_ci_jobs_pass_1.png](./imgs/submit_pr_final_commit_ci_jobs_pass_1.png)
+
+### 21. Merge your pull request
+
+ Finally `merge` your pull request changes into the `master` branch of the `strongloop/loopback-next` [repo](https://github.com/strongloop/loopback-next).
```

Comment:
Correct me if I'm wrong but I don't think people who aren't maintainers have merging rights.

Reply:
We maintainers are responsible for merging the PR.

Reply:
+1

Let's rephrase this section to make it clear that the contributor has to wait until project maintainers merge the pull request.

Also please consider adding a section explaining what to do after the pull request was merged - how to update `master` branch in `origin` to point to the same commit as the `master` branch in `upstream`.

Reply:
@bajtos , I have addressed
"
Let's rephrase this section to make it clear that the contributor has to wait until project maintainers merge the pull request.
"